File: research/nbc/query.js
query endCard(
  $id: ID
  $type: VideoType!
  $mpxGuid: String!
  $forcedMpxGuid: String
  $userId: String!
  $timeZone: String!
  $platform: SupportedPlatforms!
  $profile: JSON
  $ld: Boolean
  $oneApp: Boolean
  $app: NBCUBrands!
  $language: Languages
  $playlistMachineName: String
  $tagLine: String
) {
  endCard(
    id: $id
    type: $type
    mpxGuid: $mpxGuid
    forcedMpxGuid: $forcedMpxGuid
    userId: $userId
    timeZone: $timeZone
    platform: $platform
    profile: $profile
    ld: $ld
    oneApp: $oneApp
    app: $app
    language: $language
    playlistMachineName: $playlistMachineName
    tagLine: $tagLine
  ) {
    ...endCard
    ...endTiles
  }
}
fragment endCard on EndCard {
  ...component
  data {
    ...endCardData
  }
  analytics {
    ...endCardAnalytics
  }
}
fragment endTiles on EndTiles {
  ...component
  data {
    ...endTilesData
  }
}
fragment endCardAnalytics on EndCardAnalyticsAttributes {
  brand {
    title
  }
  recommendationType
  series
  movie
  title
  programmingType
  episodeNumber
  seasonNumber
  mpxGuid
  locked
  genre
  duration
  playlistMachineName
  endCardVariant
  endCardLogic
}
fragment component on Component {
  component
  meta
  treatments
}
fragment endCardData on EndCardData {
  ...componentData
  titleTitle
  title
  image
  description
  videoMetaData
  tagLine
  urlAlias
  buttonLabel
  mpxGuid
  permalink
  programmingType
  cuePoint
  locked
  labelBadge
  percentViewed
  whiteBrandLogo
  colorBrandLogo
  brandDisplayTitle
  mpxAccountId
  mpxAdPolicy
  resourceId
  channelId
  titleLogo
  playlistMachineName
  rating
  alternateGroupTagLine
  alternateOne {
    ...endCardAlternate
  }
  alternateTwo {
    ...endCardAlternate
  }
  entityType
  v4ID
  gradientEnd
  gradientStart
  lightPrimaryColor
  darkPrimaryColor
}
fragment endTilesData on EndTilesData {
  ...componentData
  titleKeyArt
  groupTagLine
  tileOne {
    ...endCardAlternate
  }
  tileTwo {
    ...endCardAlternate
  }
  notification {
    ...notification
  }
}
fragment notification on Notification {
  ...component
  data {
    ...componentData
    v4ID
    machineName
    headline
    headlineColor
    message
    messageColor
    logo
    logoAltText
    portraitImage
    landscapeImage
    cta {
      ...ctaLink
    }
  }
  analytics {
    entityTitle
  }
}
fragment ctaLink on CTALink {
  ...component
  data {
    ...ctaData
  }
  analytics {
    smartDynamicCTA
    smartTileLabel
    smartTileScenario
    ctaTitle
    destinationType
    destination
    brand {
      title
    }
    series
    movie
    isMovie
    videoTitle
    locked
    programmingType
    seasonNumber
    episodeNumber
    mpxGuid
    duration
    isPlaylist
    playlistMachineName
    playlistTitle
    isLive
  }
}
fragment ctaData on CTAData {
  ...componentData
  color
  text
  destinationType
  destination
  playlistMachineName
  playlistCount
  urlAlias
  isLive
}
fragment endCardAlternate on EndCardAlternate {
  ...component
  data {
    ...endCardAlternateData
  }
  analytics {
    ...endCardAnalytics
  }
}
fragment endCardAlternateData on EndCardAlternateData {
  ...componentData
  titleKeyArt
  tagLine
  playlistMachineName
  tile {
    ...videoTile
    ...seriesTile
    ...movieTile
  }
}
fragment videoTile on VideoTile {
  ...component
  data {
    ...videoItem
  }
  analytics {
    brand {
      title
    }
    series
    title
    programmingType
    episodeNumber
    seasonNumber
    mpxGuid
    locked
    duration
    genre
    movie
  }
}
fragment seriesTile on SeriesTile {
  ...component
  data {
    ...seriesItem
  }
  analytics {
    series
    genre
    brand {
      title
    }
  }
}
fragment movieTile on MovieTile {
  ...component
  data {
    ...movieItem
  }
  analytics {
    movie
    genre
    brand {
      title
    }
  }
}
fragment videoItem on VideoItem {
  ...componentData
  ...item
  tertiaryTitle
  locked
  mpxGuid
  programmingType
  episodeNumber
  seasonNumber
  airDate
  percentViewed
  permalink
  lastWatched
  duration
  genre
  rating
  lightPrimaryColor
  darkPrimaryColor
  seriesShortTitle
  movieShortTitle
  whiteBrandLogo
  colorBrandLogo
  brandDisplayTitle
  mpxAccountId
  mpxAdPolicy
  resourceId
  channelId
  rating
  externalAdvertiserId
  ariaLabel
  secondaryTitleTag
  longDescription
  ctaText
  ctaTextColor
  brandMachineName
  durationBadge
}
fragment seriesItem on SeriesItem {
  ...componentData
  ...item
  seriesName
  shortTitle
  urlAlias
  favoritedOn
  favoriteID
  posterImage
  lightPrimaryColor
  darkPrimaryColor
  whiteBrandLogo
  colorBrandLogo
  brandDisplayTitle
  landscapePreview
  portraitPreview
}
fragment movieItem on MovieItem {
  ...componentData
  ...item
  urlAlias
  favoritedOn
  favoriteID
  posterImage
  image
  lightPrimaryColor
  darkPrimaryColor
  whiteBrandLogo
  colorBrandLogo
  brandDisplayTitle
  landscapePreview
  portraitPreview
  rating
}
fragment item on Item {
  v4ID
  title
  secondaryTitle
  description
  image
  gradientStart
  gradientEnd
  labelBadge
  lastModified
}
fragment componentData on ComponentData {
  instanceID
}

File: research/nbc/query.go
package main

const query = `
query bonanzaPage(
  $id: ID
  $name: String!
  $type: EntityPageType!
  $nationalBroadcastType: String
  $userId: String!
  $platform: SupportedPlatforms!
  $device: String
  $profile: JSON
  $ld: Boolean
  $oneApp: Boolean
  $timeZone: String
  $deepLinkHandle: String
  $app: NBCUBrands!
  $nbcAffiliateName: String
  $telemundoAffiliateName: String
  $language: Languages
  $playlistMachineName: String
  $mpxGuid: String
  $authorized: Boolean
  $minimumTiles: Int
  $endCardMpxGuid: String
  $endCardTagLine: String
  $isDayZero: Boolean
) {
  bonanzaPage(
    id: $id
    name: $name
    type: $type
    nationalBroadcastType: $nationalBroadcastType
    userId: $userId
    platform: $platform
    device: $device
    profile: $profile
    ld: $ld
    oneApp: $oneApp
    timeZone: $timeZone
    deepLinkHandle: $deepLinkHandle
    app: $app
    nbcAffiliateName: $nbcAffiliateName
    telemundoAffiliateName: $telemundoAffiliateName
    language: $language
    playlistMachineName: $playlistMachineName
    mpxGuid: $mpxGuid
    authorized: $authorized
    minimumTiles: $minimumTiles
    endCardMpxGuid: $endCardMpxGuid
    endCardTagLine: $endCardTagLine
    isDayZero: $isDayZero
  ) {
    id
    pageType
    name
    data {
      sections {
         ...error
         ...videoPlayer
      }
    }
  }
}

fragment error on Error {
  error
}

fragment videoPlayer on VideoPlayer {
  data {
    player {
      v4ID
      mpxGuid
      locked
      programmingType
      title
      secondaryTitle
      tertiaryTitle
      seasonNumber
      episodeNumber
      genre
      amazonGenre
      duration
      percentViewed
      airDate
      copyright
      dayPart
      description
      shortDescription
      sunset
      keywords
      permalink
      rating
      seriesShortTitle
      seriesUrlAlias
      lightPrimaryColor
      gradientStart
      gradientEnd
      whiteBrandLogo
      colorBrandLogo
      brandDisplayTitle
      brandMachineName
      mpxAccountId
      mpxAdPolicy
      resourceId
      allowMobileWebPlayback
      ariaLabel
      startRecapTiming
      endRecapTiming
      startTeaserTiming
      endTeaserTiming
      startIntroTiming
      endIntroTiming
      cuePoint
      externalAdvertiserId
      ratingAdvisories
      tmsId
      movieShortTitle
      allowSkipButtons
      skipButtonsDuration
    }
  }
  analytics {
    tmsId
    isLongFormContent
    mpxGuid
    durationInMilliseconds
    airDate
    dayPart
    webBrandDomain
    episodeNumber
    permalink
    series
    movie
    seasonNumber
    genre
    secondaryGenre
    title
    locked
    titleTmsId
    clipCategory
    adobeVideoPlatform
    adobeContentType
    adobeBrand
    convivaAssetName
    videoBroadcast
    isOlympics
    listOfGenres
    programmingType
    rating
    ratingAdvisories
    duration
    nielsenProgen
    nielsenBrand
    nielsenSfCode
    nielsenClientId
    nielsenChannel
    ottPlatform
    league
    sport
    event
    language
    brand {
      title
    }
  }
}
`
